### PR TITLE
[#6][Backend] As a user, I can login

### LINKED ISSUE #6: [Backend] As a user, I can login
## Why

Before the user can take the survey, they have to login with their email and password first.

## Acceptance Criteria

- Calls the Login API with user email and password
  - The form should show errors if their credential is wrong
  - Navigate to the **blank** homepage if the credential is correct

## Resources

- [[Postman] Login API](https://cloudy-sunset-1396-1.postman.co/workspace/My-Workspace~27fb5d1b-f211-4dae-86d7-015bf33fc388/request/691973-7f721f33-1ef0-4751-b37f-7d1eb4f37392)
- [[Figma] Login form](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?node-id=6%3A168)
- [[Figma] Login form error](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?node-id=103%3A132)

### PR DESCRIPTION
**Pull request by rosle**
Closes https://github.com/rosle/react-nimble-survey/issues/6

## What happened 👀

- When user submits a valid credential, redirect to Home page.
- Displays the error message (from API) otherwise. 

### Import Component's test IDs issue

I've tried to import the test IDs into the integration test directly, so we don't need to have a duplicate test id definition on the test file, you can check my attempt here -> https://github.com/rosle/react-nimble-survey/pull/29

On that PR, I did:
- Customize the webpack config for integration test because to import the component, it needs to process the SVG import (for logos on that page) and also to process the typescript
- Configure the webpack alias so that we can do `import { loginScreenTestIds } from 'screens/Login';` instead of the relative path

But after making it works, there are issue with the code coverage
- The coverage for App routes are missing
- The coverage is incorrect. I tried removing the success login case and the coverage is still 100%. I guess it is because we add the Cypress's webpack preprocessor plugin and maybe it goes through all the code making it 100%. I can see a lot of hit on the line that should be missing coverage.

With all of these reason, I keep our old way of doing integration test. By duplicating it again on the integration test files.

## Proof Of Work 📹

https://28--react-nimble-survey-staging.netlify.app/sign_in

https://user-images.githubusercontent.com/6965195/187867295-661f276c-e530-47e4-83c8-b7ec3c716743.mov

### CODE REVIEW

File: cypress/e2e/Authentication/login.cy.ts [hunk 1]
```diff
@@ -1,3 +1,26 @@
+// TODO: How to reuse the component's test ids
```
Comment (rosle) [inline cypress/e2e/Authentication/login.cy.ts]: I'll put an improvement in another PR 🙏

Comment (rosle) [reply cypress/e2e/Authentication/login.cy.ts]: After adding it, it does mess with the code coverage 🥲 I'm gonna close that Pr and maybe put some summarize of the issue here 📝

File: package.json [hunk 2]
```diff
@@ -32,6 +32,10 @@
     "@cypress/webpack-dev-server": "2.0.0",
     "@nimblehq/eslint-config-nimble-react": "1.1.0",
     "@nimblehq/stylelint-config-nimble": "1.0.2",
+    "@pollyjs/adapter-node-http": "6.0.5",
+    "@pollyjs/core": "6.0.5",
```
Comment (hanam1ni) [inline package.json:36]: Interesting library 👍 I never heard of this one before

**Review by hanam1ni — APPROVED** [review]
So many test setup in the PR ✨

**Review by Lahphim — APPROVED** [review]
`persister-fs` looks nice!!

**Pull request merged**